adoc/headers/reducer.h

```diff
@@ -4,39 +4,51 @@
 // Exposition only
 template <typename T, typename BinaryOperation, int Dimensions, /* unspecified */>
 class reducer {
+public:
 
-  reducer(const reducer<T,BinaryOperation,Dimensions>&) = delete;
-  reducer<T,BinaryOperation,Dimensions>& operator(const reducer<T,BinaryOperation,Dimensions>&) = delete;
+  using value_type = T;
+  using binary_operation = BinaryOperation;
+  static constexpr int dimensions = Dimensions;
+
+  reducer(const reducer&) = delete;
+  reducer(reducer&&) = delete;
```

Comment:
Are we sure we want to make `reducer` non-movable? What's the benefit?

Reply:
If I remember correctly, we wanted to permit implementations of reducer that looked like this:
```c++
template <typename T>
struct reducer {
  T _data;
};
```
...in which case, the developer should always pass around a reference to the reducer that was created by the SYCL implementation rather than creating copies of it.

I think deleting these constructors may also stop a developer from accidentally sharing a reducer between multiple work-items (e.g. by writing the reducer out to memory).

Reply:
It was clear that the intent was to make it non-copyable and non-moveable.

(I'm not saying we shouldn't discuss it; rather, that seems like more than a bug fix or minor improvement of something not previously considered.)